Octokit/Models/Request/TeamRepositoryUpdate.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Octokit.Internal;
+using System.Diagnostics;
+using System.Globalization;
+
+namespace Octokit
+{
+    public enum PermissionType
+    {
+        [Parameter(Value = "pull")]
+        Pull,
+        [Parameter(Value = "push")]
+        Push,
+        [Parameter(Value = "admin")]
+        Admin
+    }
+
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class TeamRepositoryUpdate
+    {
+        /// <summary>
+        /// Used to add or update a team repository.
+        /// </summary>
+        public TeamRepositoryUpdate(PermissionType permission)
+        {
+            Permission = permission;
+        }
+
+        /// <summary>
+        /// The permission to grant the team on this repository.
+        /// </summary>
+        [Parameter(Key = "permission")]
```

Comment:
You shouldn't need to specify `[Parameter]` tag except in unusual cases.  The default serialization will handle converting between github API format and octokit.net's more c# representation

eg:
`lowercase_separated_with_underscores` <=> `LowercaseSeparatedWithUnderscores`

Only when a property name on our side doesn't match the above conversion, do you need to explicitly define a `[Parameter]` tag